### PR TITLE
#6066: Extract json deDot feature from metricbeat/http module to libbeat/common

### PR DESCRIPTION
**Pull request by christiangalsterer**

### CODE REVIEW

**Comment by christiangalsterer** [comment]
@ruflin 
Hi Nicolas,

as discussed here the PR for extracting the de dot feature for json into libbeats.
Please let me you of any feedback/changes.

File: libbeat/common/_meta/test/json_map_dedot.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
         "key_with_dot_l2": 1,
         "key_with_multiple_dots_l2": 2,
         "key_without_dot_l2": {
-            "key_with_dot_l2": 3,
-            "key_with_multiple_dots_l2": 4
+            "key_with_dot_l3": 3,
```
Comment (ruflin) [inline libbeat/common/_meta/test/json_map_dedot.json]: Let's call the directory `testdata` which is more idiomatic for Golang. The `_meta` data directory is specific to our modules where we automatically generate some information, we don't need it here I think.

**Comment by christiangalsterer** [comment]
Moved the test data to testdata folder

File: libbeat/common/_meta/test/json_map_dedot.json (hunk 1 above)
Comment (christiangalsterer) [reply libbeat/common/_meta/test/json_map_dedot.json]: Moved the test data to testdata folder

**Comment by ruflin** [comment]
jenkins, test it

**Comment by ruflin** [comment]
jenkins, test it

**Pull request merged**